neo4j/_async_compat/concurrency.py

```diff
@@ -108,7 +121,16 @@ async def acquire(self, blocking=True, timeout=-1):
             await self._acquire(me)
             return True
         try:
-            await asyncio.wait_for(self._acquire(me), timeout)
+            fut = asyncio.ensure_future(self._acquire(me))
+            try:
+                await wait_for(fut, timeout)
+            except asyncio.CancelledError:
+                already_finished = not fut.cancel()
+                if already_finished:
+                    # Too late to cancel the acquisition.
+                    # This can only happen in Python 3.7's asyncio
```

Comment:
```suggestion
                    # This can only happen in Python 3.7's asyncio 
                    # as well as in our custom wait_for shim.
```